### ISSUE TITLE
find_essential input gradients are wrong for fewer than 9 points (zero for 5): the SVD backward drops the null-space rows

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4855]
On `main` (`9c39a109`, macOS arm64 CPU, torch 2.14.0; also seen at N=5 on torch 2.9.1), the gradient of `find_essential` with respect to the points disagrees with central differences whenever `N < 9`. For the minimal case `N = 5`, which RANSAC uses, it is exactly zero:

```python
import torch
from kornia.geometry.epipolar import find_essential

for n in (5, 6, 8, 9, 12):
    g = torch.Generator().manual_seed(0)
    p = torch.rand(1, n, 2, generator=g, dtype=torch.float64).requires_grad_()
    q = torch.rand(1, n, 2, generator=g, dtype=torch.float64)
    f = lambda p: (find_essential(p, q).nan_to_num() ** 3).sum()
    (grad,) = torch.autograd.grad(f(p), p)
    fd = torch.zeros(2 * n, dtype=torch.float64)
    for k in range(2 * n):
        d = torch.zeros(2 * n, dtype=torch.float64); d[k] = 1e-6; d = d.view_as(p)
        fd[k] = (f(p.detach() + d) - f(p.detach() - d)) / 2e-6
    print(n, (grad.flatten() - fd).abs().max().item(), fd.abs().max().item())
# N=5: 2.01 vs 2.01 (autograd is 0), N=6: 29.5 vs 33.8, N=8: 13.4 vs 16.1, N=9: 0.561 vs 23.4, N=12: 9.7e-05 vs 1.53
```

`_null_to_Nister_solution_script` takes the null space as the last four rows of `Vh` from `torch.linalg.svd(X)`, with `X` of shape `(N, 9)` and full matrices. PyTorch's SVD backward only differentiates the first `min(N, 9)` singular vectors and silently drops the gradient that reaches the others. The check `X = torch.rand(5, 9, requires_grad=True); _, _, Vh = torch.linalg.svd(X); Vh[-4:].pow(3).sum().backward()` leaves `X.grad` at exactly 0. For `N = 5` every null-space row is dropped. For `6 <= N < 9` some of them are. A differentiable pipeline that backpropagates through a minimal 5-point solve therefore gets no gradient to the correspondences.

A fix needs a null-space parametrization whose derivative is defined. The null space of a 5×9 matrix has no unique basis, but the final candidates do not depend on which basis is used. Candidates are implicit differentiation of the candidates, or computing the null space from a square matrix such as `X^T X` with an explicit gauge. No test covers `find_essential` gradients today.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by VanniLeonardo** [comment]
I will take this one. A maintainer will need to assign me.

**Comment by ducha-aiki** [comment]
@VanniLeonardo it's yours, and draft #4860 is the PR for it. We can't assign outside contributors, so this comment is the claim. Mark #4860 ready for review when it is done. Drafts are not reviewed.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
